micrometer-observation/src/test/java/io/micrometer/observation/docs/ObservationDocumentationTests.java

```diff
@@ -65,6 +67,17 @@ void iaeShouldBeThrownWhenDocumentedObservationHasOverriddenDefaultConventionBut
             .hasMessageContaining("but you have provided an incompatible one of type");
     }
 
+    @Test
+    void whenGetNameIsNotOverridden_thenNameShouldBeTakenFromDefaultConvention() {
+        ObservationRegistry registry = observationRegistry();
+        Observation.Context context = new Observation.Context();
+
+        TestConventionObservation.OVERRIDDEN.observation(registry, () -> context).start().stop();
+
+        then(context.getName()).isEqualTo("one");
```

Comment:
Moving this to a separate issue to consider to not block progress on the nullability migration.